### ISSUE TITLE
Remote SSH: Ambiguous message for connection error

### ISSUE THREAD

**Issue by tuxedocat** [microsoft/vscode-remote-release#6887]
Issue Type: <b>Bug</b>

- VSCode Version: Version: 1.68.1 (Universal) Commit: 30d9c6cd9483b2cc586687151bcbcd635f373630
- Local OS Version: macOS 12.4 (21F79)
- Remote OS Version: Debian 4.19.235-1 (2022-03-17) x86_64 GNU/Linux
- Remote Extension/Connection Type: SSH
  <details>
  
  <summary>
  The instance is on GCP, and connected through Cloud IAP using gcloud tool
  </summary>
  
  ```sshconfig
  Host ****
    HostName compute.****
    IdentityFile /Users/****/.ssh/google_compute_engine
    CheckHostIP no
    HashKnownHosts no
    HostKeyAlias compute.****
    IdentitiesOnly yes
    StrictHostKeyChecking yes
    UserKnownHostsFile /Users/****/.ssh/google_compute_known_hosts
    ProxyCommand /opt/homebrew/bin/python3 -S /Users/****/google-cloud-sdk/lib/gcloud.py compute start-iap-tunnel INSTANCE_NAME %p --listen-on-stdin --project=PROJECT_ID --zone=ZONE --verbosity=warning
    ProxyUseFdpass no
    User ****
  ```
  
  </details>

- Logs:
```
[13:17:57.439] Log Level: 2
[13:17:57.440] remote-ssh@0.82.1
[13:17:57.440] darwin arm64
[13:17:57.456] SSH Resolver called for "ssh-remote+****", attempt 1
[13:17:57.457] "remote.SSH.useLocalServer": true
[13:17:57.457] "remote.SSH.path": undefined
[13:17:57.457] "remote.SSH.configFile": undefined
[13:17:57.458] "remote.SSH.useFlock": true
[13:17:57.458] "remote.SSH.lockfilesInTmp": false
[13:17:57.458] "remote.SSH.localServerDownload": auto
[13:17:57.458] "remote.SSH.remoteServerListenOnSocket": false
[13:17:57.459] "remote.SSH.showLoginTerminal": true
[13:17:57.459] "remote.SSH.defaultExtensions": []
[13:17:57.459] "remote.SSH.loglevel": 2
[13:17:57.459] "remote.SSH.enableDynamicForwarding": true
[13:17:57.460] "remote.SSH.enableRemoteCommand": false
[13:17:57.460] "remote.SSH.serverPickPortsFromRange": {}
[13:17:57.460] "remote.SSH.serverInstallPath": {}
[13:17:57.465] SSH Resolver called for host: ****
[13:17:57.465] Setting up SSH remote "****"
[13:17:57.467] Acquiring local install lock: /var/folders/wq/0hb_5kdn457_71qzj025r7k5kw21d8/T/vscode-remote-ssh-b39f68d8-install.lock
[13:17:57.469] Looking for existing server data file at /Users/****/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-***-********-0.82.1/data.json
[13:17:57.475] Resolver error: TypeError: Cannot read properties of undefined (reading 'replace')
	at Object.a [as sanitizeConnectionToken] (/Users/****/.vscode/extensions/ms-vscode-remote.remote-ssh-0.82.1/out/extension.js:1:691667)
	at Object.C [as findRunningServer] (/Users/****/.vscode/extensions/ms-vscode-remote.remote-ssh-0.82.1/out/extension.js:1:626477)
	at async /Users/****/.vscode/extensions/ms-vscode-remote.remote-ssh-0.82.1/out/extension.js:1:622445
	at async T (/Users/****/.vscode/extensions/ms-vscode-remote.remote-ssh-0.82.1/out/extension.js:1:619286)
	at async Object.t.resolveWithLocalServer (/Users/****/.vscode/extensions/ms-vscode-remote.remote-ssh-0.82.1/out/extension.js:1:622395)
	at async Object.t.resolve (/Users/****/.vscode/extensions/ms-vscode-remote.remote-ssh-0.82.1/out/extension.js:1:644769)
	at async /Users/****/.vscode/extensions/ms-vscode-remote.remote-ssh-0.82.1/out/extension.js:1:726943
[13:17:57.479] ------
```

Steps to Reproduce:

1. Connect remote instance via ssh as usual
2. (somehow error below happens)
    ![Screen Shot 2022-07-04 at 13 19 05](https://user-images.githubusercontent.com/1410242/177082898-dc49f90c-1da0-441e-83f3-390a08ae16d2.jpg)
3. Current workaround is to delete `/Users/****/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-***-********-0.82.1/` then reload the editor

## tl;dr

The message shown in the dialogue is not helpful to solve the issue.
It is helpful that the extension automatically resolve this kind of error.

---

Does this issue occur when you try this locally?: Yes

Extension version: 0.82.1
VS Code version: Code 1.68.1 (Universal) (30d9c6cd9483b2cc586687151bcbcd635f373630, 2022-06-14T12:52:13.188Z)
OS version: Darwin arm64 21.5.0
Restricted Mode: No
Connection to 'SSH: ****' could not be established  Cannot read properties of undefined (reading 'replace')
Remote OS version: Linux x64 4.19.0-20-cloud-amd64

<details>

<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|32.00GB (0.33GB free)|
|Process Argv|--crash-reporter-id a2cea4fd-097b-4f1d-af2c-97aea603b43b|
|Screen Reader|no|
|VM|0%|

Connection to 'SSH: ****' could not be established  Cannot read properties of undefined (reading 'replace')

|Item|Value|
|---|---|
|Remote|SSH: **** |
|OS|Linux x64 4.19.0-20-cloud-amd64|
|CPUs|Intel(R) Xeon(R) CPU @ 2.00GHz (8 x 2000)|
|Memory (System)|29.45GB (10.20GB free)|
|VM|0%|

</details>

**Comment by KikeM** [comment]
The workaround worked for me, thanks!

**Comment by tuxedocat** [comment]
Bit late and this seems stale issue, but FYI, disabling shared connection makes retrying connection in each opened SSH-remote window work fine.

```json
"remote.SSH.useLocalServer": false
```

**Comment by KikeM** [comment]
So, as I said I tried the solution mentioned above.

It did work in terms of not showing the error anymore, but my SSH connection became much more faulty, and sometimes it would not even connect.

I have set it back to `true` and now it is working fine again.
